x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizer.java

```diff
@@ -123,13 +128,21 @@ public PhysicalPlan apply(PhysicalPlan plan) {
                     // otherwise expect a Fragment
                     if (child instanceof FragmentExec fragmentExec) {
                         var logicalFragment = fragmentExec.fragment();
+                        var skipProject = false;
                         // no need for projection when dealing with aggs
                         if (logicalFragment instanceof Aggregate) {
-                            attributes.clear();
+                            skipProject = true;
                         }
                         var selectAll = projectAll.get();
-                        if (attributes.isEmpty() == false || selectAll) {
+                        if (skipProject == false) {
                             var output = selectAll ? exec.child().output() : new ArrayList<>(attributes);
+                            // a physical plan cannot be empty - return a constant instead
+                            if (output.isEmpty()) {
+                                var alias = new Alias(logicalFragment.source(), "<all-fields-projected>", null, Literal.NULL, null, true);
```

Comment:
Send as data a dummy null block which will contain as many entries as the count.
To avoid a clash with existing projections, use the synthetic flag.